samples/LiveEffect/src/main/cpp/LiveEffectEngine.h

```diff
@@ -14,24 +14,25 @@
  * limitations under the License.
  */
 
-#ifndef OBOE_ECHOAUDIOENGINE_H
-#define OBOE_ECHOAUDIOENGINE_H
+#ifndef OBOE_LIVEEFFECTENGINE_H
+#define OBOE_LIVEEFFECTENGINE_H
 
-#include <thread>
+#include <oboe/Oboe.h>
 #include <jni.h>
+#include <thread>
 #include <string>
-#include "audio_common.h"
-#include "AudioEffect.h"
 
+constexpr int kMonoChannelCount = 1;
+constexpr int kStereoChannelCount = 2;
 const int32_t kLoopbackSampleRate = 48000;
```

Comment:
The sample rate shouldn't be hardcoded, it should be obtained from the output stream, then used to create the inputstream.

Reply:
The usage would be better:
  app asks for the sample rate
  app checks for the sample rate really granted
  if not granted with different rate, do something else

The reasoning would be: if app wants to mix audio, it would already have certain sample rate in preference; it may not be exactly the same as the platform's default sample rate. If platform could support app's sample rate, app would not have to re-sample. It is more app friendly rather than leave all work to app. In this sample, it assert that the sample rate is what asked for. What do you think?